sklearn/cluster/k_means_.py

```diff
@@ -313,12 +320,25 @@ def k_means(X, n_clusters, init='k-means++', precompute_distances='auto',
     x_squared_norms = row_norms(X, squared=True)
 
     best_labels, best_inertia, best_centers = None, None, None
+    if n_clusters == 1:
```

Comment:
But we will still have to compute the distances for the first iteration (I think)

Anyhow, this is not that important, just a nitpick.